composer/core/state.py

```diff
@@ -175,12 +176,18 @@ def train_dataloader(self, dataloader: types.DataLoader):
         self.train_data = DataSpec(dataloader)
 
     @property
-    def eval_dataloader(self):
-        return self.eval_data.dataloader
-
-    @eval_dataloader.setter
-    def eval_dataloader(self, dataloader: types.DataLoader):
-        self.eval_data = DataSpec(dataloader)
+    def evaluators(self):
+        return self.evaluator_data
+
+    @evaluators.setter
+    def evaluators(self, evaluators: Sequence[types.Evaluator]):
+        for evaluator in evaluators:
```

Comment:
```suggestion
        self._evaluators[:] = ensure_tuple(optimizers)
```
This will ensure that the evaluators is updated in-place in case if something has a reference to it.